### ISSUE TITLE
Yubikey multiple usage of OTPs

### ISSUE THREAD

**Issue by gholamiali** [hopshadoop/hopsworks#303]
currently, when a user authenticates using a Yubikey device, Yubikey table in the database is not updated. 

One possible cause is the loop in the model EJBs  that resemble Yubikey and User tables.

**Comment by jimdowling** [comment]
What is the symptom of this problem? That the same OTP can be used more than once? That login doesn't work when it should or works when it shouldn't?

**Comment by gholamiali** [comment]
Correct. An OTP can be reused several times because Yubikey table does not get updated despite the update method in CauthRealm returns "1" which means success. But actual commit is not done. This line:

https://github.com/biobankcloud/CauthRealm/blob/master/src/main/java/se/kth/bbc/crealm/CustomAuthRealm.java#L462


 So in summary, when an OTP is used once, it should not be possible to reuse it again.

**Comment by Gumuservi** [comment]
Indeed, a little while ago I downloaded Wordpress Shield WAF only for it's acceptance of YubiKey OTP. I saw the loginscreen saved the first used OTP so tried to login with it and it worked. Allthough the Shield dev fixed the issue I am now using the addon with Yubikey AND Google authenticator.
